plugin/updater_gpu/src/gpu_builder.cu

```diff
@@ -327,12 +327,12 @@ void GPUBuilder::Sort(int level) {
 }
 
 void GPUBuilder::Update(const std::vector<bst_gpair> &gpair, DMatrix *p_fmat,
-                        RegTree *p_tree) {
+                        RegTree &tree) {
```

Comment:
Ok, I got it.
I will iterate tomorrow in the morning and edit this PR accordingly.